website/src/components/UI/Affix.tsx

```diff
@@ -0,0 +1,99 @@
+import type { CSSProperties, ReactNode, FC } from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import { getDomStyle } from '../../utils';
+
+interface Props {
+  style: CSSProperties;
+  children: ReactNode;
+}
+
+const AffixContent = styled.div`
+`;
+
+const getPositionStyle = (
+  hasCssSticky: boolean,
+  defaultStyles: CSSProperties = {},
+):CSSProperties => {
+  const { width } = defaultStyles;
+  const positionStyle: CSSProperties = hasCssSticky ? {
+    position: 'sticky',
+    marginLeft: `-${width}px`,
+    display: 'inline-block',
+    float: 'left',
+  } : {
+    position: 'absolute',
+  };
+
+  return {
+    ...positionStyle,
+    ...defaultStyles,
+  };
+};
+
+const Affix: FC<Props> = (props) => {
+  const { style, children } = props;
+  const [hasCssSticky, SetHasCssSticky] = useState(true);
+  useEffect(() => {
+    SetHasCssSticky(CSS.supports('position', 'sticky'));
+  });
+
+  const positionStyle = getPositionStyle(hasCssSticky, style);
+
+  const defaultHeight = parseInt(`${style.top}`, 10);
```

Comment:
`${style.top}` really?

Reply:
<img width="766" alt="image" src="https://user-images.githubusercontent.com/23497322/169700308-0a61ce55-4fe6-4aa2-a01a-396c39a7a0c1.png">

like this, really is `props.style.top` 😂

Reply:
I mean, is it a number or a string? Is it possible to write it in a way that is not a template string?